lib/galaxy/webapps/galaxy/api/group_users.py

```diff
@@ -14,45 +16,95 @@
     BaseGalaxyAPIController,
     depends,
 )
+from galaxy.schema.schema import (
+    GroupUserListModel,
+    GroupUserModel,
+)
+from . import (
+    depends,
+    DependsOnTrans,
+    Router,
+)
 
 log = logging.getLogger(__name__)
 
+router = Router(tags=["group_users"])
+
+GroupIDParam: EncodedDatabaseIdField = Path(..., title="GroupID", description="The ID of the group")
+
+UserIDParam: EncodedDatabaseIdField = Path(..., title="UserID", description="The ID of the user")
+
+def group_user_to_model(trans, encoded_group_id, user):
+    encoded_user_id = trans.security.encode_id(user['id'])
+    url = trans.url_builder("group_user", group_id=encoded_group_id, id=encoded_user_id)
+    return GroupUserModel(id=encoded_user_id, email=user['email'], url=url)
 
-class GroupUsersAPIController(BaseGalaxyAPIController):
-    manager = depends(GroupUsersManager)
+@router.cbv
+class FastAPIGroupUsers:
+    manager: GroupUsersManager = depends(GroupUsersManager)
 
-    @require_admin
-    @expose_api
-    def index(self, trans: ProvidesAppContext, group_id: EncodedDatabaseIdField, **kwd):
+    # @require_admin
+    # @expose_api
+    @router.get("/api/groups/{group_id}/users", require_admin=True, summary="Displays a collection (list) of groups.")
+    def index(
+        self, trans: ProvidesAppContext = DependsOnTrans, group_id: EncodedDatabaseIdField = GroupIDParam
+    ) -> GroupUserListModel:
         """
         GET /api/groups/{encoded_group_id}/users
         Displays a collection (list) of groups.
         """
-        return self.manager.index(trans, group_id)
+        group_users = self.manager.index(trans, group_id)
+        return GroupUserListModel(__root__=[group_user_to_model(trans, group_id, gr) for gr in group_users])
 
-    @require_admin
-    @expose_api
-    def show(self, trans: ProvidesAppContext, id: EncodedDatabaseIdField, group_id: EncodedDatabaseIdField, **kwd):
+    # @require_admin
+    # @expose_api
+    @router.get(
+        "/api/groups/{group_id}/users/{id}",
+        name="group_user",
+        require_admin=True,
+        summary="Displays information about a group user.",
+    )
```

Comment:
These were available as both `/api/groups/{group_id}/users/{id}` and `/api/groups/{group_id}/user/{id}`.
You can either repeat the decorator with the additional path, or (and these are bonus point ;)) add functionality to the router class (https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/webapps/galaxy/api/__init__.py#L253) to register the additional paths, so we could use something like this:
```suggestion
    @router.get(
        "/api/groups/{group_id}/user/{id}",
        alias="/api/groups/{group_id}/users/{id}",
        name="group_user",
        require_admin=True,
        summary="Displays information about a group user.",
    )
```

Reply:
Updated.